compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt

```diff
@@ -65,6 +77,36 @@ internal class PlatformWindowContext {
         )
     }
 
+    /**
+     * Converts the given [Rect] from the coordinate space of the container window to the coordinate
+     * space of the root UIWindow in which hierarchy with the container window resides.
+     */
+    fun convertContainerWindowRectToRootWindowCGRect(rect: Rect): CValue<CGRect> {
+        val windowContainer = _windowContainer ?: return CGRectZero.readValue()
+
+        val windowContainerWindow = windowContainer.window
+
+        fun UIWindow.density(): Density {
+            return Density(screen.scale.toFloat())
+        }
+
+        return if (windowContainerWindow == null) {
+            if (windowContainer is UIWindow) {
+                // windowContainer is the root window itself
+                rect.toDpRect(windowContainer.density()).asCGRect()
+            } else {
+                // windowContainer is not attached to any window
+                CGRectZero.readValue()
+            }
+        } else {
+            val cgRect = rect
+                .toDpRect(windowContainerWindow.density())
+                .asCGRect()
+
+            windowContainer.convertRect(cgRect, toView = windowContainerWindow)
```

Comment:
Can we extract generic `convertRect` function like we have for `convertPoint`?

Reply:
We'll have a lot of back-and-forth dp-to-pp bouncing then. The whole purpose of this API is to return `CGRect`

Reply:
```kt
    fun convertWindowRect(boundsInWindow: Rect, toView: UIView): Rect {
        val windowContainer = _windowContainer ?: return boundsInWindow
        return convertRect(
            rect = boundsInWindow,
            fromView = windowContainer,
            toView = toView
        )
    }

    private fun convertRect(rect: Rect, fromView: UIView, toView: UIView): Rect {
        return if (fromView != toView) {
            val density = fromView.systemDensity
            fromView.convertRect(
                rect = rect.toDpRect(density).asCGRect(),
                toView = toView,
            ).useContents {
                asDpRect().toRect(density)
            }
        } else {
            rect
        }
    }
```

Reply:
> back-and-forth dp-to-pp bouncing then

Yep, but since it's type-safe and clear, it's not a big deal

Reply:
I've added it, keep in mind that it's a _lot_ of recalculations in accessibility scope. Accessibility resolves all the visible bounds for elements to determine their reachability.